share/spack/gitlab/cloud_pipelines/stacks/e4s-mac/spack.yaml

```diff
@@ -0,0 +1,64 @@
+spack:
+  view: false
+  concretization: separately
+
+  config:
+    concretizer: clingo
+    install_tree:
+      root: $SPACK_ROOT/opt/spack
+      padded_length: 512
+      projections:
+        all: '{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+
+  packages:
+    all:
+      compiler: [apple-clang@13.0.0]
+      target: [m1]
+
+  definitions:
+
+  - easy_specs:
+    - zlib
+
+  - arch:
+    - '%apple-clang@13.0.0 target=m1'
+
+
+  specs:
+
+  - matrix:
+    - - $easy_specs
+    - - $arch
+
+  mirrors: { "mirror": "s3://spack-binaries-develop/e4s-mac" }
+
+  gitlab-ci:
+
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd ${SPACK_CONCRETE_ENV_DIR}
+      - spack env activate --without-view .
+      - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+      - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+      - spack -d ci rebuild > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
```

Comment:
I guess we'll see if there are any issues with this incantation on mac, I haven't tried it, but maybe fine.